src/components/InputGroup/InputGroup.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import styled from 'styled-components';
+import { prop } from 'ramda';
+
+import { SpaceSizes } from '../../theme';
+import { getFormStyle, getRadii, getSpace } from '../../utils';
+import { Input, Select } from '../forms';
+import { Icon } from '../Icon';
+import { IconButton } from '../IconButton';
+import { Inline } from '../layout';
+import { InputGroupProps } from './InputGroup.types';
+import { InlineProps } from '../layout/Inline/Inline';
+
+const InputGroupContainer = styled(Inline)<InputGroupProps>`
+  border: ${getFormStyle('borderWidth')} solid ${getFormStyle('borderColor')};
+  border-radius: ${getRadii('default')};
+
+  & > *:first-child > .ssc__control {
+    border: 0px;
+    box-shadow: none;
+  }
+  & > & {
+    border: none;
+    border-radius: 0px;
+  }
+  & > * {
+    border: none;
+    border-radius: 0px;
+    height: ${getFormStyle('fieldHeight')};
+    ${(props) =>
+      props.divider &&
+      `border-right: ${getFormStyle('borderWidth')(props)} solid ${getFormStyle(
+        'borderColor',
+      )(props)};`}
+  }
+  & > *:first-child {
+    border-radius: ${getRadii('default')} 0 0 ${getRadii('default')};
+  }
+  & > *:last-child {
+    border-right: 0px;
+    border-radius: 0 ${getRadii('default')} ${getRadii('default')} 0;
+  }
+  & > svg {
+    flex-grow: 0;
+    flex-shrink: 0;
+    margin: 0 ${getSpace(SpaceSizes.sm)} !important;
+    display: block;
+  }
+`;
+
+const InputGroup: React.FC<InputGroupProps & InlineProps> = ({
+  children,
+  divider,
+  ...inlineProps
+}) => {
+  const ALLOWED_CHILDREN = [Select, Input, Icon, IconButton, InputGroup];
```

Comment:
I would remove IconButton (it is icon in the circle, it probably won't work with this) and add a normal `Button` instead. Let's also add `Password` and `MultiValueInput`